src/can.c

```diff
@@ -109,8 +163,29 @@ static void thingset_can_report_rx_cb(const struct device *dev, struct can_frame
     const struct thingset_can *ts_can = user_data;
     uint16_t data_id = THINGSET_CAN_DATA_ID_GET(frame->id);
     uint8_t source_addr = THINGSET_CAN_SOURCE_GET(frame->id);
-
-    ts_can->report_rx_cb(data_id, frame->data, can_dlc_to_bytes(frame->dlc), source_addr);
+#ifdef CONFIG_THINGSET_CAN_PACKETIZED_REPORTS_RX
+    if (THINGSET_CAN_PACKETIZED_REPORT(frame->id)) {
+        struct can_rx_buffer *buffer = NULL;
+        if ((buffer = thingset_can_get_rx_buf(source_addr)) != NULL) {
```

Comment:
I suggest to use a `net_buf`s here (probably `net_buf_simple` are fine).

Unfortunately, the documentation is not particularly great: https://docs.zephyrproject.org/latest/connectivity/networking/api/net_buf.html#structnet__buf__simple

You may have to look into the implementation to understand the differences between the functions.

Reply:
As you can see, I've ended up using the full fat `net_buf`; I needed `user_data` for the context and the built-in linked list node was a useful replacement for rolling my own.